core/sync.js

```diff
@@ -95,51 +98,88 @@ class Sync {
   // - push if only changes from the fs are applied to the remote cozy
   // - full for the full synchronization of the both sides
   async start(mode /*: SyncMode */) /*: Promise<*> */ {
-    this.stopped = false
+    if (this.started && (await this.started)) return
 
-    let sidePromises = []
-    if (mode !== 'pull') {
-      await this.local.start()
-      sidePromises.push(this.local.watcher.running)
-    }
-    if (mode !== 'push') {
-      const { running, started } = this.remote.start()
-      sidePromises.push(running)
-      await started
-    }
-    await new Promise(
-      async function(resolve, reject) {
-        Promise.all(sidePromises).catch(err => reject(err))
-        try {
-          // eslint-disable-next-line no-constant-condition
-          while (true) {
-            await this.sync()
-          }
-        } catch (err) {
-          reject(err)
+    let runningResolve, runningReject
+    this.running = new Promise((resolve, reject) => {
+      runningResolve = resolve
+      runningReject = reject
+    })
+
+    this.started = new Promise(async (resolve, reject) => {
+      if (this.stopRequested) reject()
+
+      log.info('Starting Sync...')
+
+      try {
+        let sidePromises = []
+        if (mode !== 'pull') {
```

Comment:
Maybe it is time to remove the push and pull modes. I don't think they are still used.

Reply:
After removing it I wonder if this is actually something we want to do or if we could use that to give users the option to use Cozy Desktop as a backup software :thinking:.
I'll create another PR to discuss this.